service/history/mutableStateBuilder.go

```diff
@@ -524,23 +491,15 @@ func (e *mutableStateBuilder) UpdateCurrentVersion(
 		return nil
 	}
 
-	// TODO when NDC is fully rolled out remove this block
-	//  since event local namespace workflow will have version history
-	if version != common.EmptyVersion {
-		err := serviceerror.NewInternal("cannot update current version of local namespace workflow to version other than empty version")
-		e.logError(err.Error())
-		return err
-	}
+	// TODO: All mutable state should have versioned histories.  Even local namespace should have it to allow for
+	// re-replication of local namespaces to other clusters.
+	// We probably need an error if mutableState does not have versioned history.
```

Comment:
can we make the namespace by default "global" & with one target replication cluster?

Reply:
This might break existing deployments of Temporal.  Could be super risky change it this point.  We might consider this after V1.